### PR TITLE
[CDAP-9290] File Browser add collapsed breadcrumb

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-9290

Breadcrumb will have a collapsed view once you are more than 3 levels deep.

### CODE REVIEW

File: cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js [hunk 1]
```diff
@@ -65,32 +75,103 @@ export default class FilePath extends Component {
...
+        <span className="path-divider">/</span>
+        {breadcrumb}
+      </div>
+    );
+  }
+
+  render() {
+    if (this.props.loading) {
+      return (
+        <div className="file-path-container">
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js]: Not sure if we need to show spinner for the breadcrumb. Can we show one for the entire content assuming we use we have one end point for all content

File: cdap-ui/app/cdap/components/FileBrowser/FilePath/FilePath.scss [hunk 2]
```diff
@@ -36,4 +37,35 @@ $path-color: #333333;
     margin-left: 5px;
     margin-right: 5px;
   }
+
+  .collapsed-dropdown,
+  .paths {
+    display: inline-block;
+  }
+
+  .collapsed-dropdown {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/FilePath/FilePath.scss:47]: Might want to give a padding or have ellipsis on longer folder names.

Noticed this one
screen shot 2017-04-23 at 10 42 55 pm

File: cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js [hunk 3]
```diff
@@ -65,32 +75,103 @@ export default class FilePath extends Component {
     });
   }
 
-  render() {
+  renderCollapsedDropdown(collapsedLinks) {
     return (
-      <div className="file-path-container">
+      <div className="collapsed-dropdown">
+        <Dropdown
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js]: Is it possible to add to `UncontrolledComponents` and just use that? We could avoid the `isOpen, toggle` stuff from being added to the state.

**Review by ajainarayanan — COMMENTED** [review]
Minor comments.

File: cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js]: I'll add this in a separate PR then

**Review by ajainarayanan — APPROVED** [review]
👍 LGTM

**Comment by bdmogal** [comment]
LGTM. I think the dropdown on the ellipsis can be made much clearer, but let's do it separately because what's in this PR is per design.

**Pull request merged**